pcdsdevices/epics_motor.py

```diff
@@ -1523,6 +1523,35 @@ def __init__(self, prefix='', *, tip_pv, tilt_pv, **kwargs):
         super().__init__(prefix, **kwargs)
 
 
+class SmarActPicoscale(SmarAct):
+    """
+    Class for encoded SmarAct motors controller via the MCS2 controller
+    which use the Picoscale sensor heads for encoded positions.
+    Child of SmarAct device.
+    """
+    # configuration settings and readbacks
+    pico_present = Cpt(EpicsSignalRO, ':PS_PRESENT', kind='config')
+    pico_exists = Cpt(EpicsSignalRO, ':HAVE_PS', kind='config')
+    pico_valid = Cpt(EpicsSignalRO, ':PS_DATA_VALID', kind='config')
+    pico_sig_qual = Cpt(EpicsSignalRO, ':PS_SIG_QUALITY', kind='config')
+
+    # Validation that Picoscale is working properly
+    pico_enable = Cpt(EpicsSignalRO, ':PS_ENABLE_RBV', kind='normal')
+    pico_stable = FCpt(EpicsSignalRO, '{self._ioc_base}:PS:STABLE', kind='normal')
+
+    # Diagnostic information from PS controller
+    pico_name = FCpt(EpicsSignalRO, '{self._ioc_base}:PS:LOCATOR',
+                     kind='config')
+    pico_wmin = FCpt(EpicsSignalRO, '{self._ioc_base}:PS:WORKING_MIN',
+                     kind='config')
+    pico_wmax = FCpt(EpicsSignalRO, '{self._ioc_base}:PS:WORKING_MAX',
+                     kind='config')
+
```

Comment:
Do you want to add any of the PVs related to the picoscale adjustment? I think these would be nice to have to attempt auto-alignment for remote recovery.

Reply:
I thought about pulling those through, but I haven't had a situation where I've actually seen those flags _change_. But that might just be the nature of what I'm staring at when debugging and nudging the stages that are being ornery. 

From what I can tell, `<IOC_BASE>:PS:CUR_ADJ_STATE` and `<IOC_BASE>:PS:REQ_ADJ_STATE` are always `disabled` in the FTL. Likewise, `CURR_ADJ_PROGRESS` and `ADJ_DONE` are always `0`.

Maybe I need to go mess around with it more with a `camonitor` on

Reply:
That is to be expected when you are not actively calibrating. 
`REQ_ADJ_STATE` allows you to set the Picoscale into "Manual" or "Auto" adjustment, same as the SmarAct GUI. 
`CUR_ADJ_STATE` is the current adjustment state. 
`CURR_ADJ_PROGRESS` is an estimate of the progress of the auto adjustment cycle. 
`ADJ_DONE` means... you're done! :)

Reply:
TIL I didn't have to connect via USB to the picoscale every time ...

I'll add these as config PVs then as it is very convenient to have.

Users probably _still_ want to be in the FTL for this, as you want to walk the stage to the front for dead path calc, but this is honestly great to have.

Reply:
I've seen it fail when the SmarAct software succeeds, so there are perhaps some bugs to sort out there. But It's better than nothing.

Reply:
So I just tried this in OL-MDL 4-6, and it worked, but it caused some unexpected motor move timeout errors while doing so, which is interesting?
![image](https://github.com/pcdshub/pcdsdevices/assets/149725219/1e8680f5-266e-4efe-80e1-f25f540ad35f)

Possible bug or expected behavior from how `MoveStatus` works? Stages were _not_ moving and had already concluded their movements before the adjustment began. Position RBV was also frozen during this time until the adjustment finished and `pico_stable` and `pico_valid` switched back to their True states.

Error did not persist, btw, and I could move it just fine after adjustment concluded.

Reply:
@aberges-SLAC Can you explain a little more how this occurred? During the auto-adjustment, you should not move the motor _at all_.

Reply:
Exactly, no movement commands were processing _before_ auto-adjustment was triggered. But it threw these errors. Not sure if it just crashed the `MoveStatus` based on an associated flag?

Reply:
OK, re-read your comment more carefully. I think I understand now.
Were the errors thrown during adjustment? Or after? 
If during, then that _might_ make sense if the picoscale stops responding during auto-adjustment.

Reply:
During, which lines up with my other observations.  Glad to know I didn't bork it :]